pkgs/development/libraries/qtkeychain/default.nix

```diff
@@ -16,16 +18,26 @@ stdenv.mkDerivation rec {
     sha256 = "04v6ymkw7qd1pf9lwijgqrl89w2hhsnqgz7dm4cdrh8i8dffpn52";
   };
 
-  cmakeFlags = [ "-DQT_TRANSLATIONS_DIR=share/qt/translations" ];
+  cmakeFlags = [ "-DQT_TRANSLATIONS_DIR=share/qt/translations" ]
+    ++ stdenv.lib.optional stdenv.isDarwin [
+       # correctly detect the compiler
+       # for details see cmake --help-policy CMP0025
+       "-DCMAKE_POLICY_DEFAULT_CMP0025=NEW"
```

Comment:
Would it make sense to set this globally? I think this is the behavior we always want but it might not make a difference for most CMake projects.

Reply:
I think so. At the least setting `CMAKE_POLICY_WARNING_CMP0025` would be helpful for identifying this issue in other packages.

Reply:
Thanks! See #37015 for the PR making this the default.

Reply:
So after doing this, we still occasionally get issues like in:

https://hydra.nixos.org/build/73579953/nixlog/1

Any ideas how to fix it?